examples/platform/k32w/app/support/Memconfig.cpp

```diff
@@ -0,0 +1,181 @@
+typedef struct A_BLOCK_LINK
+{
+    struct A_BLOCK_LINK * pxNextFreeBlock; /*<< The next free block in the list. */
+    size_t xBlockSize;                     /*<< The size of the free block. */
+} BlockLink_t;
+
+/* The size of the structure placed at the beginning of each allocated memory
+block must by correctly byte aligned. */
+static const size_t xHeapStructSize =
+    (sizeof(BlockLink_t) + ((size_t)(portBYTE_ALIGNMENT - 1))) & ~((size_t) portBYTE_ALIGNMENT_MASK);
+
+/* Gets set to the top bit of an size_t type.  When this bit in the xBlockSize
+member of an BlockLink_t structure is set then the block belongs to the
+application.  When the bit is free the block is still part of the free heap
+space. */
+static size_t xBlockAllocatedBit = ((size_t) 1) << ((sizeof(size_t) * heapBITS_PER_BYTE) - 1);
+
+extern "C" {
+
+/* xPortMallocUsableSize relies on heap4 implementation.
+It returns the size of an allocated block and it is
+called by __wrap_realloc.
+Thus it is validated in __wrap_realloc function that the allocated size
+of the old_ptr is smaller than the allocated size of new_ptr */
+size_t xPortMallocUsableSize(void * pv)
+{
+    uint8_t * puc = (uint8_t *) pv;
+    BlockLink_t * pxLink;
+    void * voidp;
+    size_t sz = 0;
+
+    if (pv != NULL)
+    {
+        vTaskSuspendAll();
+        {
+            /* The memory being checked will have an BlockLink_t structure immediately
+            before it. */
+            puc -= xHeapStructSize;
+
+            /* This casting is to keep the compiler from issuing warnings. */
+            voidp  = (void *) puc;
+            pxLink = (BlockLink_t *) voidp;
+
+            /* Check if the block is actually allocated. */
+            configASSERT((pxLink->xBlockSize & xBlockAllocatedBit) != 0);
+            configASSERT(pxLink->pxNextFreeBlock == NULL);
+
+            sz = (pxLink->xBlockSize & ~xBlockAllocatedBit) - xHeapStructSize;
+        }
+        (void) xTaskResumeAll();
+    }
+
+    return sz;
+}
+
+void * __wrap_malloc(size_t size)
+{
+    return pvPortMalloc(size);
+}
+
+void __wrap_free(void * ptr)
+{
+    vPortFree(ptr);
+}
+
+void * __wrap_calloc(size_t num, size_t size)
+{
+    size_t total_size = num * size;
+
+    // Handle overflow from (num * size)
+    if ((size != 0) && ((total_size / size) != num))
+    {
+        return nullptr;
+    }
+
+    void * ptr = pvPortMalloc(total_size);
+    if (ptr)
+    {
+        memset(ptr, 0, total_size);
+    }
+
+    return ptr;
+}
+
+void * __wrap_realloc(void * ptr, size_t new_size)
+{
+    if (new_size)
+    {
+        void * new_ptr = pvPortMalloc(new_size);
+        if (new_ptr)
+        {
+            /* If passed-in `ptr` was NULL, `realloc(NULL, size)`
+             must be equivalent to `malloc(size)`. */
+            if (ptr)
+            {
+                size_t old_ptr_size = xPortMallocUsableSize(ptr);
+                if (new_size > old_ptr_size)
+                {
+                    memset(new_ptr, 0, new_size);
```

Comment:
Why does this zero the memory first? Especially when we're going to overwrite |old_ptr_size| zeroes.

Reply:
Modified with @tcarmelveilleux suggestion